inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -146,26 +146,26 @@ def remove(self, parameters: ApplicationRemoveSourceParameters) -> None:
         # Remove the file under /etc/apt/sources.list.d
         try:
             if (
-                os.path.sep in parameters.file_name
-                or parameters.file_name == ".."
-                or parameters.file_name == "."
+                os.path.sep in parameters.source_list_file_name
+                or parameters.source_list_file_name == ".."
+                or parameters.source_list_file_name == "."
             ):
-                raise SourceError(f"Invalid file name: {parameters.file_name}")
+                raise SourceError(f"Invalid file name: {parameters.source_list_file_name}")
```

Comment:
**Suggestion:** Instead of checking for the presence of path separators and specific relative paths, use a more robust method to sanitize the filename. [security]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            from pathlib import Path
            sanitized_filename = Path(parameters.source_list_file_name).name
            if sanitized_filename in ("..", ".", ""):
                raise SourceError(f"Invalid file name: {parameters.source_list_file_name}")
```